addons/account/models/account_payment.py

```diff
@@ -869,240 +848,86 @@ def create(self, vals_list):
-            vals['move_type'] = 'entry'
+            # Hack to add a custom line.
+            linecomplete_line_vals_list.append(vals.pop('line_ids', None))
 
-        payments = super(AccountPayment, self.with_context(is_payment=True)).create([{
-            'name': False,
-            **vals,
-        } for vals in vals_list])
+        payments = super().create(vals_list)
 
         for i, (pay, vals) in enumerate(zip(payments, vals_list)):
-            # Write payment_id on the journal entry plus the fields being stored in both models but having the same
-            # name, e.g. partner_bank_id. The ORM is currently not able to perform such synchronization and make things
-            # more difficult by creating related fields on the fly to handle the _inherits.
-            # Then, when partner_bank_id is in vals, the key is consumed by account.payment but is never written on
-            # account.move.
-            to_write = {'payment_id': pay.id, 'name': False}
-            for k, v in vals_list[i].items():
-                if k in self._fields and self._fields[k].store and k in pay.move_id._fields and pay.move_id._fields[k].store:
-                    to_write[k] = v
-
-            if 'line_ids' not in vals_list[i]:
-                to_write['line_ids'] = [
-                    Command.create(line_vals)
-                    for line_vals in pay._prepare_move_line_default_vals(
-                        write_off_line_vals=write_off_line_vals_list[i],
-                        force_balance=force_balance_vals_list[i],
-                    )
-                ]
-            with self.env.protecting(self.env['account.move']._get_protected_vals(vals, pay)):
-                pay.move_id.write(to_write)
-            self.env.add_to_compute(self.env['account.move']._fields['name'], pay.move_id)
-
-        # We need to reset the cached name, since it was recomputed on the delegate account.move model
-        payments.invalidate_recordset(fnames=['name'])
-        return payments.with_env(self.env)  # clear the context
+            if (
+                write_off_line_vals_list[i] is not None
+                or force_balance_vals_list[i] is not None
+                or linecomplete_line_vals_list[i] is not None
+            ):
+                pay._synchronize_to_moves(
+                    write_off_line_vals=write_off_line_vals_list[i],
+                    force_balance=force_balance_vals_list[i],
+                    line_ids=linecomplete_line_vals_list[i],
+                )
+                # propagate the related fields to the move as it is being created after the payment
+                if move_vals := {
+                    fname: value
+                    for fname, value in vals.items()
+                    if self._fields[fname].related and (self._fields[fname].related or '').split('.')[0] == 'move_id'
+                }:
+                    pay.move_id.write(move_vals)
```

Comment:
ok, after having read _synchronize_to_moves, I better understand but I don't like it much: can't we move that related fields propagation inside synchronize_to_moves, and do it directly during the move creation ? I don't see any reason why to keep that in here :thinking: 


what do you think?

Reply:
You have in mind that there is still a sync being done.
While that would be ideal, some influential people decided that it was not necessary.
We will probably come back on that in the near future...

Reply:
ok, no big deal anyway